torch/distributed/pipelining/stage.py

```diff
@@ -1256,33 +1264,46 @@ def __init__(
         stage_index: int,
         num_stages: int,
         device: torch.device,
-        input_args: Union[torch.Tensor, Tuple[torch.Tensor, ...]],
+        input_args: Optional[Union[torch.Tensor, Tuple[torch.Tensor, ...]]] = None,
```

Comment:
Shall we take this chance to make the signature right?
It should always be `Tuple[torch.Tensor, ...]` even if there is only 1 tensor input, not `Union`.

Reply:
hmm, do we have a pattern to follow here? i thought i remembered other torch methods accepting a single item or a list/tuple of items in some cases.  But i would agree with your suggestion. I worry a bit about BC breaking, but i guess that's probably not in widespread use yet.

Reply:
i can work on this but i would prefer to keep it a separate PR